sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/NoSuchItemException.scala

```diff
@@ -291,3 +292,7 @@ class NoSuchIndexException private(
       messageParameters = Map.empty[String, String])
   }
 }
+
+class NoSuchVariableException(variable: Seq[String])
```

Comment:
Sorry I may miss the discussion result before. So we will keep adding new exception types even if we have unique error class name now?

Reply:
I was following: "when in Rome, do as the Romans do". Can change it whole sale for all new errors.

Reply:
Done